specutils/analysis/width.py

```diff
@@ -0,0 +1,212 @@
+    """
+    Compute the true full width half max of the spectrum.
+    
+    This makes no assumptions about the shape of the spectrum (e.g. whether it
+    is Gaussian). It finds the maximum of the spectrum, and then locates the
+    point closest to half max on either side of the maximum, and
+    measures the distance between them. This will be calculated over the
+    regions, if they are specified. 
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object over which the width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion` or list of `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the FWHM value.
+
+    Returns
+    -------
+    whm : `~astropy.units.Quantity` or list (based on region input)
+        Full width of the signal at half max
+    """
+    return _computation_wrapper(_compute_fwhm, spectrum, region)
+
+
+def _compute_gaussian_fwhm(spectrum, region=None):
+    """
+    Estimate the full width of the spectrum at half max.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object overwhich the width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the FWHM value.
+
+    Returns
+    -------
+    gaussian_fwhm : `~astropy.units.Quantity` or list (based on region input)
+        Approximate full width of the signal at half max
+
+    Notes
+    -----
+    This is a helper function for the above `gaussian_fwhm()` method.
+
+    """
+
+    if region is not None:
+        calc_spectrum = region.extract(spectrum)
+    else:
+        calc_spectrum = spectrum
+
+    flux = calc_spectrum.flux
+    spectrum = calc_spectrum.spectral_axis
+
+    dx = spectrum - np.mean(spectrum)
+    fwhm = 2 * np.sqrt(np.sum((dx * dx) * flux, axis=-1) / np.sum(flux, axis=-1))
+
+    return fwhm
+
+
+def _compute_gaussian_sigma_width(spectrum, region=None):
+    """
+    Estimate the full width of the spectrum based on an approximate value of
+    sigma.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object overwhich the width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the gaussian sigma width.
+
+    Returns
+    -------
+    full_width : `~astropy.units.Quantity` or list (based on region input)
+        Approximate full width of the signal
+
+    Notes
+    -----
+    This is a helper function for the above `gaussian_sigma_width()` method.
+
+    """
+
+    fwhm = _compute_gaussian_fwhm(spectrum, region)
+    sigma = fwhm * gaussian_fwhm_to_sigma
+
+    return sigma * 2
+
+
+def _arghalf(array, halfval):
+    """
+    Find index of the item closest to the half max
+    """
+    return np.abs(array - halfval).argmin()
+
+
+def _compute_single_fwhm(flux, spectrum):
```

Comment:
(readability nitpick, don't need to both with it if you don't want to): 
`spectrum` is a bit confusing of a variable name because it seems like you're talking about the `Spectrum1D`, when actually this is the `spectral_axis`.  Maybe `specax`?